### DISCUSSION TITLE
How can i use html preview in openvscode server?

### DISCUSSION THREAD

**Discussion by jiungbaek** [gitpod-io/openvscode-server#367] [Q&A]
How can i use html preview in openvscode server?

preview extention can not use in my openvscode server.

ex) run openvscode server in 192.168.0.1 server and my web browser is in my desktop.

i am using Live Preview extension

![image](https://user-images.githubusercontent.com/7424183/174023912-26a442f0-8af9-4891-955a-777b0134f419.png)

**Comment by filiptronicek** [comment]
Hey there @jiungbaek, unfortunately, serving localhost URLs is not possible, see https://github.com/gitpod-io/openvscode-server/discussions/335 for more.

**Reply by jeanp413** [reply]
That issue is about something different

**Comment by jeanp413** [comment]
@jiungbaek For webviews to work you need to:
1.  Access the vscode frontend over `https` if you are not in localhost e.g. `https://192.168.0.1:<port>`
2. For preview extension or any other extension that needs some service in a different port:
    - Ideal fix, you need to configure your own reverse proxy, so different origins/subdomains are mapped to a port in the server (this is what gitpod/codespaces do) e.g. 
      - `https://gitpodio-templatetypesc-sskm55x3tyr.ws-us47.gitpod.io/` (this is mapped to openvscode-server port) 
      -  `https://3000-gitpodio-templatetypesc-sskm55x3tyr.ws-us47.gitpod.io` this is mapped to port `3000` in the server where an http server is running
     - Create an ssh tunnel to the port your service is running `ssh -L 9000:192.168.0.1:9000 user@192.168.0.1` and then access it using `http://localhost:9000`

**Reply by jiungbaek** [reply]
My openvscodeserver runs in k8s, I did port forwarding like https://3000-myopenvscode.xxxxx.com

we decided office preview plugin to preview html.

also, we are using tabnine, but tabnine hub in plugin is not work like this.

Does Tabnine Hub need to be set up like this for it to work?
